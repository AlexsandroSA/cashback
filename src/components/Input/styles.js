import styled from 'styled-components';

export const InputWrapper = styled.div`
    margin-bottom: 18px;
`;

export const Label = styled.label`
    margin-bottom: 0;
`;

export const Input = styled.input`
    margin-bottom: 0;

    &:disabled {
        cursor: not-allowed;
        background: #dddddd;
        color: #000;
    }
`;

export const FeedbackError = styled.span`
    display: block;
    font-size: 12px;
    color: #f00;
    text-align: right;
`;
