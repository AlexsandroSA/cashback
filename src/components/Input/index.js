import React from 'react';
import PropTypes from 'prop-types';
import InputMask from 'react-input-mask';

import { InputWrapper, Label, Input, FeedbackError } from './styles';

function InputBase({
    id,
    testId,
    name,
    label,
    value,
    mask,
    disabled,
    htmlType,
    touched,
    error,
    onBlur,
    onChange,
}) {
    const hasMask = !!mask;
    return (
        <InputWrapper>
            <Label htmlFor={id}>{label}</Label>
            {hasMask ? (
                <InputMask
                    disabled={disabled}
                    autoComplete="off"
                    data-testid={testId}
                    id={id}
                    name={name}
                    type={htmlType}
                    onBlur={onBlur}
                    onChange={onChange}
                    value={value}
                    mask={mask}
                />
            ): (
                <Input
                    disabled={disabled}
                    autoComplete="off"
                    data-testid={testId}
                    id={id}
                    name={name}
                    type={htmlType}
                    onBlur={onBlur}
                    onChange={onChange}
                    value={value}
                />
            )}
            <FeedbackError data-testid={`${testId || id}-error`}>{touched && error}</FeedbackError>
        </InputWrapper>
    )
}

InputBase.prototype = {
    id: PropTypes.string,
    name: PropTypes.string.isRequired,
    label: PropTypes.string,
    value: PropTypes.string,
    htmlType: PropTypes.string,
    error: PropTypes.string,
    touched: PropTypes.bool,
    disabled: PropTypes.bool,
    onBlur: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
};

InputBase.defaultProps = {
    label: '',
    value: '',
    error: '',
    id: 'input',
    testId: 'input',
    htmlType: 'text',
    disabled: false,
};

export default InputBase;
