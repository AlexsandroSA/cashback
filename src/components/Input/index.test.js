import React from 'react';
import { shallow } from 'enzyme';

import Input from './index';

let defaultProps;

describe('Input', () => {
  beforeEach(() => {
    defaultProps = {
      onBlur: jest.fn(),
      onChange: jest.fn(),
      name: 'test',
    };
  });

  it('should be defined', () => {
    expect(Input).toBeDefined();
  });

  it('renders correctly', () => {
        expect(Input).toMatchSnapshot();
    });

  it('should render whit props', () => {
    const wrapper = shallow(<Input {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render whit mask', () => {
    const wrapper = shallow(<Input mask='99/99/9999' {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
