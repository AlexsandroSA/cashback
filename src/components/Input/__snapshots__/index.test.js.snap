// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `[Function]`;

exports[`Input should render whit mask 1`] = `ShallowWrapper {}`;

exports[`Input should render whit props 1`] = `ShallowWrapper {}`;
